pip/pip-344.md

```diff
@@ -0,0 +1,87 @@
+# PIP-344: Correct the behavior of the public API pulsarClient.getPartitionsForTopic(topicName)
+
+# Background knowledge
+
+**Topic auto-creation**
+- The partitioned topic auto-creation is dependent on `pulsarClient.getPartitionsForTopic`
+    - It triggers partitioned metadata creation by `pulsarClient.getPartitionsForTopic`
+    - And triggers the topic partition creation by producers' registration and consumers' registration.
```

Comment:
What do you mean here? I don't think the partitioned topic auto-creation is dependent on `PulsarClient.getPartitionsForTopic`. From the following description, it seems that the `PulsarClient.getPartitionsForTopic` API could create the partition metadata automatically.

Reply:
So it should not be "is dependent on". "A is dependent on B" means A cannot work without B. The description is confusing, we can just retain the following sentence "When calling ..." and remove the description above, which should be more clear.